x-pack/plugins/ml/public/application/components/import_export_jobs/import_jobs_flyout/import_jobs_flyout.tsx

```diff
@@ -0,0 +1,493 @@
+                  className="file-datavisualizer-file-picker"
+                />
+              </div>
+
+              {showFileReadError && <CannotReadFileCallout />}
+
+              {totalJobsRead > 0 && jobType !== null && (
+                <>
+                  <EuiSpacer size="l" />
+                  {jobType === 'anomaly-detector' && (
+                    <FormattedMessage
+                      id="xpack.ml.importExport.importFlyout.selectedFiles.ad"
+                      defaultMessage="{num} anomaly detection {num, plural, one {job} other {jobs}} read from file"
+                      values={{ num: totalJobsRead }}
+                    />
+                  )}
+
+                  {jobType === 'data-frame-analytics' && (
+                    <FormattedMessage
+                      id="xpack.ml.importExport.importFlyout.selectedFiles.dfa"
+                      defaultMessage="{num} data frame analytics {num, plural, one {job} other {jobs}} read from file"
+                      values={{ num: totalJobsRead }}
+                    />
+                  )}
+
+                  <EuiSpacer size="m" />
+
+                  <CannotImportJobsCallout
+                    jobs={skippedJobs}
+                    autoExpand={jobIds.length === 0 || skippedJobs.length === 1}
+                  />
+
+                  <FormattedMessage
+                    id="xpack.ml.importExport.importFlyout.importableFiles"
+                    defaultMessage="Import {num, plural, one {# job} other {# jobs}}"
+                    values={{ num: jobIds.length }}
+                  />
+                  <EuiSpacer size="m" />
+
+                  {jobIds.map((jobId, i) => (
+                    <div key={i}>
+                      <EuiPanel hasBorder={true}>
+                        <EuiFlexGroup>
+                          <EuiFlexItem>
+                            <EuiFormRow
+                              error={jobId.invalidMessage}
+                              isInvalid={jobId.invalidMessage.length > 0}
+                            >
+                              <EuiFieldText
+                                prepend={i18n.translate(
+                                  'xpack.ml.importExport.importFlyout.jobId',
+                                  {
+                                    defaultMessage: 'Job ID',
+                                  }
+                                )}
+                                disabled={importing}
+                                compressed={true}
+                                value={jobId.id}
+                                onChange={(e) => renameJob(e.target.value, i)}
+                                isInvalid={jobId.valid === false}
+                              />
+                            </EuiFormRow>
+                          </EuiFlexItem>
+                          <EuiFlexItem grow={false}>
+                            <DeleteJobButton index={i} />
```

Comment:
I used the index here because the alternative is to pass the job ID, look through the array to find the index and then use it to splice the array.
Also the job ids can change and there could be two jobs with the same ID if the user is halfway through renaming them.
Using the index seemed like the safer and faster thing to do.